client/payment-methods/index.js

```diff
@@ -121,9 +121,27 @@ const PaymentMethods = () => {
 	const availablePaymentMethodIds = useGetAvailablePaymentMethodIds();
 
 	// We filter link payment method since this will be displayed in other section (express checkout).
-	const availableMethods = availablePaymentMethodIds
-		.filter( ( id ) => 'link' !== id )
-		.map( ( methodId ) => methodsConfiguration[ methodId ] );
+	// We further split the available methods into pay later and non-pay later methods to sort them in the required order later.
+	const availableNonPayLaterMethods = availablePaymentMethodIds.filter(
+		( id ) =>
+			'link' !== id &&
+			'card' !== id &&
+			! methodsConfiguration[ id ].allows_pay_later
+	);
+
+	const availablePayLaterMethods = availablePaymentMethodIds.filter(
+		( id ) => 'link' !== id && methodsConfiguration[ id ].allows_pay_later
+	);
+
+	const orderedAvailablePaymentMethodIds = [
+		'card',
+		...availablePayLaterMethods,
+		...availableNonPayLaterMethods,
+	];
+
+	const availableMethods = orderedAvailablePaymentMethodIds.map(
```

Comment:
That's a bit over-engineering IMO. IIUC, we want to add BNPL at the end of `availableMethods` list, or?

Reply:
Gotcha, then it's a good structure.